deepchem/molnet/load_function/material_datasets.py

```diff
@@ -0,0 +1,359 @@
+
+  """
+
+  # Featurize
+  logger.info("About to featurize band gap dataset.")
+  my_tasks = ['gap expt']  # machine learning targets
+
+  # Get DeepChem data directory if needed
+  if data_dir is None:
+    data_dir = DEFAULT_DIR
+  if save_dir is None:
+    save_dir = DEFAULT_DIR
+
+  # Check for str args to featurizer and splitter
+  if isinstance(featurizer, str):
+    featurizer = DEFAULT_FEATURIZERS[featurizer](**featurizer_kwargs)
+  elif issubclass(featurizer, Featurizer):
+    featurizer = featurizer(**featurizer_kwargs)
+
+  if isinstance(splitter, str):
+    splitter = DEFAULT_SPLITTERS[splitter]()
+  elif issubclass(splitter, Splitter):
+    splitter = splitter()
+
+  # Reload from disk
+  if reload:
+    featurizer_name = str(featurizer.__class__.__name__)
+    splitter_name = str(splitter.__class__.__name__)
+    save_folder = os.path.join(save_dir, "bandgap-featurized", featurizer_name,
+                               splitter_name)
+
+    loaded, all_dataset, transformers = deepchem.utils.save.load_dataset_from_disk(
+        save_folder)
+    if loaded:
+      return my_tasks, all_dataset, transformers
+
+  # First type of supported featurizers
+  supported_featurizers = ['ElementPropertyFingerprint'
+                          ]  # type: List[Featurizer]
+
+  # Load .tar.gz file
+  if featurizer.__class__.__name__ in supported_featurizers:
+    dataset_file = os.path.join(data_dir, 'expt_gap.tar.gz')
+    deepchem.utils.untargz_file(dataset_file, dest_dir=data_dir)
+    dataset_file = os.path.join(data_dir, 'expt_gap.json')
+
+    if not os.path.exists(dataset_file):
+      deepchem.utils.download_url(url=BANDGAP_URL, dest_dir=data_dir)
+      deepchem.utils.untargz_file(
+          os.path.join(data_dir, 'expt_gap.tar.gz'), data_dir)
+
+    # Changer loader to match featurizer and data file type
+    loader = deepchem.data.JsonLoader(
+        tasks=my_tasks,
+        feature_field="composition",
+        label_field="gap expt",
+        featurizer=featurizer)
+
+  # Featurize dataset
+  dataset = loader.create_dataset(dataset_file)
+
+  train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(
+      dataset, **splitter_kwargs)
+
+  # Initialize transformers
+  transformers = [
+      DEFAULT_TRANSFORMERS[t](dataset=dataset, **transformer_kwargs[t])
+      if isinstance(t, str) else t(
+          dataset=dataset, **transformer_kwargs[str(t.__name__)])
+      for t in transformers
+  ]
+
+  for transformer in transformers:
+    train_dataset = transformer.transform(train_dataset)
+    valid_dataset = transformer.transform(valid_dataset)
+    test_dataset = transformer.transform(test_dataset)
+
+  if reload:  # save to disk
+    deepchem.utils.save.save_dataset_to_disk(
+        save_folder, train_dataset, valid_dataset, test_dataset, transformers)
+
+  return my_tasks, (train_dataset, valid_dataset, test_dataset), transformers
+
+
+def load_perovskite(
```

Comment:
I think it is better to split each function into one file. We create new directory like `deepchem/molnet/load_function/material_dataset`, and then put these files like `load_bandgap.py` `load_perovskite.py`and on it. (And then, it is better to split tests)